docs/index.md

```diff
@@ -17,11 +17,22 @@ access PatternPal from `View > Other Windows > PatternPal Extension`.
 
 You can open the main view of the extension through `View > Other Windows > PatternPal Extension`.
 
+### Demo
+
+The following video offers a walkthrough for the usage of PatternPal: 
+https://www.youtube.com/watch?v=nl327X8Ixyk
+
+### Recognizer
+
+This module allows the user to select the desired design patterns from a list to recognize for. The user has the option to either recognize patterns from their contemporary active document, or one of their projects. 
+
+When the user clicks on the 'Analyze' button, the extension attempts to recognize the selected design patterns in the active document or selected project and displays its analysis. The analysis depicts for each recognized pattern a general score bar depicting how confident PatternPal is in the accuracy of the implementation of the given design pattern. A score for the implementation of each of the selected patterns' components is also given. All the requirements for each of the pattern's components are also displayed, together with an indicator whether the given requirement has been fulfilled.
```

Comment:
Hier moet nog genoemd worden dat je kan klikken op de namen naast de "icoontjes" om ernaar toe te gaan in de code.